wire-ios/Wire-iOS/Sources/AuthenticatedRouter.swift

```diff
@@ -86,11 +92,21 @@ final class AuthenticatedRouter: NSObject {
     private func notifyFeatureChange(_ note: Notification) {
         guard
             let change = note.object as? FeatureRepository.FeatureChange,
-            let alert = UIAlertController.fromFeatureChange(change, acknowledger: featureRepositoryProvider.featureRepository)
+            let alert = change.hasFurtherActions
+                ? UIAlertController.fromFeatureChangeWithActions(change,
+                                                                 acknowledger: featureRepositoryProvider.featureRepository,
+                                                                 actionsHandler: featureChangeActionsHandler)
+                : UIAlertController.fromFeatureChange(change,
+                                                      acknowledger: featureRepositoryProvider.featureRepository)
         else {
             return
         }
 
+        if case .e2eIEnabled(gracePeriod: let gracePeriod) = change, let gracePeriod {
+            let endOfGracePeriod = Date.now.addingTimeInterval(gracePeriod)
```

Comment:
if I am not mistaken, the gracePeriod is always nil here, see other comment from recurring action